doc/man3/SSL_CTX_set_alpn_select_cb.pod

```diff
@@ -49,16 +49,24 @@ SSL_select_next_proto, SSL_get0_alpn_selected, SSL_get0_next_proto_negotiated
 
 =head1 DESCRIPTION
 
-SSL_CTX_set_alpn_protos() and SSL_set_alpn_protos() are used by the client to
-set the list of protocols available to be negotiated. The B<protos> must be in
-protocol-list format, described below. The length of B<protos> is specified in
-B<protos_len>. Setting B<protos_len> to 0 clears any existing list of ALPN
-protocols and no ALPN extension will be sent to the server.
+SSL_CTX_set_alpn_protos() and SSL_set_alpn_protos() can be used by both
+clients and servers. For clients, they set the list of protocols available
+to be negotiated. For servers, if no callback is registered via
+SSL_CTX_set_alpn_select_cb(), they define the list of supported protocols.
+During the TLS handshake, the server will automatically select the first
+protocol from the client's list that matches any entry in its own list.
```

Comment:
I'd suggest that in the presence of the `SSL_OP_CIPHER_SERVER_PREFERENCE` option then we alternatively select the first protocol from the server's list that matches one in the client's list.